app/models/katello/upstream_pool.rb

```diff
@@ -4,9 +4,14 @@ class UpstreamPool < OpenStruct
 
     class << self
       def fetch_pools(params)
-        response = CP_POOL.get(params: cp_request_params.deep_merge(params))
-
+        pool_ids = pool_id_params(params.delete("pool_ids"))
+        included_fields = params.delete("quantities_only") ? minimal_fields : all_fields
```

Comment:
fetch_pools is starting to get a bit long. maybe included_fields should be another method in the class and you can give it the same treatment as pool_id_params:

```
included_fields: included_fields(params.delete(:quantities_only))
```

Another refactor could be pull the hash returned (with :pool and :total keys) into `response_to_pools` as it might be more appropriate there

I'm not explicitly asking for these changes but you might consider it -  it'll make changing this class easier in the future and arguably more testable w/ smaller units